jaxb-ri/xjc/src/test/java/com/sun/tools/xjc/api/Driver.java

```diff
@@ -36,41 +34,49 @@
 
 /**
  * Tests the XJC API.
- * 
+ *
  * @author
  *     Kohsuke Kawaguchi (kohsuke.kawaguchi@sun.com)
  */
 public class Driver {
-    
+
     private static class ErrorReceiverImpl extends ConsoleErrorReporter implements ErrorListener {}
-    
+
     /** Use StAX? Otherwise use SAX. */
-    public BooleanOption stax = new BooleanOption("-stax");
+    private boolean stax;
     /** Generate code and dump it to screen? */
-    public BooleanOption code = new BooleanOption("-code");
-    
+    private boolean code;
+
     private File[] files;
-    
-    private Driver( String[] args ) throws CmdLineException {
-        CmdLineParser p = new CmdLineParser();
-        p.addOptionClass(this);
-        p.parse(args);
-        
-        List<String> files = p.getArguments();
+
+    private Driver( String[] args ) {
+        List<String> files = new ArrayList<>();
+        for (String arg : args) {
+            if ("-stax".equals(arg)) {
```

Comment:
another place where switch may be a bit better

Reply:
this is a test and this as such is also more readable then using switch with using continue instead of break to force breaking the loop.